### ISSUE TITLE
Unclear license

### ISSUE THREAD

**Issue by irtimmer** [moonlight-stream/moonlight-common-c#30]
To my surprise I discovered that unlike the moonlight-common repository their is no license included in the moonlight-common-c repository. While the usage in moonlight-chrome and moonlight-ios and the inclusion 
of ported code from moonlight-common at least indicates its available under a GPLv3 (compatible) license it would probably be better if the license is clearly stated. Just in case someone wants to use it in a new port ;)

**Comment by cgutman** [comment]
Good point. Fixed in https://github.com/moonlight-stream/moonlight-common-c/commit/417c8be124c7e235b5e46c59e073aa2a6b41dea9

**Comment by Benjamin-Dobell** [comment]
@cgutman Any chance we could get this under a more permissive license? Ideally one that's iOS compatible.

MIT or Apache would be my preference, but I think LGPL _may_ work, it's a bit unclear.

Of course, won't argue if not, your work, you rights, your license. Just thought it'd be worth asking.

**Comment by cgutman** [comment]
I don't mind relicensing my contributions, but we'd need to decide on a license and probably get @irtimmer, @d3m3vilurr, and @mrb113 's consent for their portions.

**Comment by d3m3vilurr** [comment]
I hate iOS AppStore :(
- GPLv3 not compatible in iOS AppStore. (so Apple can remove moonlight-ios in AppStore)
- GPLv2/LGPL maybe can use in AppStore, but some sections are unclear;
  VLC was multiple changes their lib and app licenses.
  And cannot back to appstore until changing dual license under MPLv2 and GPLv2 or later.
  (at now, lib is [LGPLv2.1 or later][1], app is [bi license][2])
  
Most cases I like Apache and MIT. but in this case, if we choice lower infectable license, we can have a chance about reduce return contribution.
If @cgutman and @irtimmer want all moonlight projects to remain `GPL` and want to receive their modification and fixing, we can't relicense to these licenses.

Anyway, I'm ok to relicense `LGPLv2.1 or later` or `MPLv2`, but not Apache, MIT.

[1]: http://www.videolan.org/press/lgpl.html
[2]: http://www.videolan.org/vlc/download-ios.html

**Comment by Benjamin-Dobell** [comment]
> GPLv3 not compatible in iOS AppStore. (so Apple can remove moonlight-ios in AppStore)

If there are just the four of you and you agree the binaries (build artefacts) themselves aren't GPL licensed, then it's not a problem - as copyright holders you can do whatever you want. It's only an issue when you start getting lots of contributors as _theoretically_ they'd complain you're violating the GPL on _their_ contributions by publishing to the iOS app store - I'm really not sure why anyone would do that though.

Being GPL is more a problem for third-parties - say for example I wanted to make my own app that incorporates this library as just a component of a larger app, and put it on the app store, I wouldn't be allowed to. You guys still could though (again assuming all contributors agree), as you're the original rights holders.

**Comment by d3m3vilurr** [comment]
No. Definitely not. GPL have guarantee how to provide the binary, source to the **user** and how to apply a license to derivatives works. And this rule allows binaries reversing.
A problem is these rules can conflict to AppStore copyright.
So they can remove apps from their store whatever complaints from license holders.
GPLv3 have more aggressive rules between GPLv2, and probably AppStore also can ban GPLv2 too.

Your problem is another issue. GNU want to derivative works should apply GPL. And want to [reduce LGPL][1] Because, If not, developers can hide fixing, modification and improvement whatever OSS project or not.
I basically agree on this, just not all. So I commented `MPLv2` or `LGPLv2.1`.
But this relicenses make lost some GPL's benefit.

[1]: https://www.gnu.org/licenses/why-not-lgpl.en.html

**Comment by Benjamin-Dobell** [comment]
> No. Definitely not. GPL have guarantee how to provide the binary, source to the user and how to apply a license to derivatives works. And this rule allows binaries reversing.
A problem is these rules can conflict to AppStore copyright.
So they can remove apps from their store whatever complaints from license holders.
GPLv3 have more aggressive rules between GPLv2, and probably AppStore also can ban GPLv2 too.

Wait... what?

You're the rights holder. You can license the software however you please. As I said, just don't license the binaries as GPL. It's just the same as the other projects dual licensing, they can do that because they're the copyright holder. Just as you can as rights holder offer proprietary licenses and GPL, like many businesses do. You can do _whatever you want_, it's simply your _users_ (i.e. me) that can't; because we've accepted your copyright licensed work under the terms of the GPL - which means _our_ work and therefore _our_ binaries need to be licensed under GPL. So Apple can reject them.

> Your problem is another issue. GNU want to derivative works should apply GPL. And want to reduce LGPL Because, If not, developers can hide fixing, modification and improvement whatever OSS project or not.

Yes, that's true. GPL is as a copyleft license. That on it's own isn't an issue. As you've stated it, just means third-parties must also license their works under GPL. That's reasonable, it's your project, you can decide what you want.

The point is that your _users_ can't submit binaries to the App Store because any binaries _we_ produce are GPL'd - your's are not necessarily, they're whatever you want them to be; again you need consent of _all_ contributors.

If you want to continue to distribute with a copyleft license (i.e GPL) then that's your call, I'm just making a request. I _think_ LGPL is okay on the app store, admittedly I'm not certain; everything else I've written I'm 100% certain about.

**Comment by d3m3vilurr** [comment]
> You can license the software however you please. As I said, just don't license the binaries as GPL. It's just the same as the other projects dual licensing, they can do that because they're the copyright holder.

it's little complicated. we should decide source code level license for the next contribution. or need signing about dual license agreements from contributors like Linux kernel.

> If you want to continue to distribute ...

I understand, and I already agreed to change LGPLv2.1 or MPLv2 ;)
Just I said, Apple make f**ked to GNU community and I can't understand this company.

> See https ...

In my country, copyright law requiring a complaint from the license holder.
But basically, this is a legal issue and I'm not a lawyer. in this case, only `lawyer` can say `okay`.
And Apple still can choice to remove GPL app whatever complaint.
`GPL conflict` mean also conflict Apple side either.
Just they didn't do that.

PS. @Benjamin-Dobell and looks no more need conversation about Apple problem. isn't it? ;)

PS2. @cgutman please, just choice next license then receive agree/disagree.
that's easier.

**Comment by 3unjee** [comment]
Greetings @cgutman and others, any news on a more permissive licence for this module ? LGPLv2/v3 would be great !
